### PR TITLE
Updating the Frontend Docs

### LINKED ISSUE #98: Update page information to new front-end steps. Customising Player Pages
The newly added pages under `EpicGames/PixelStreamingInfrastructure/Frontend/Docs`  all contain information relevant to the previous version of PS Infra/front end.

Update the page info and clear out any old information that no longer applies.
Add extra information that may be missing.
Delete the TO-DO from the top of all the pages.

### PR DESCRIPTION
**Pull request by StomyPX**
Not ready for merging yet (not for a little while yet). Preliminary pull request made so there's somewhere for review feedback to live.

Resolves #98

### CODE REVIEW

File: Frontend/Docs/Communicating from UE5 to the Player Page.md [hunk 1]
```diff
@@ -6,26 +6,29 @@ You can make your Unreal Engine application emit custom events to all connected
 
 To set this up:
 
-1.  In your Unreal Engine application, any time you want to emit an event to the player page, use the **Pixel Streaming > Send Pixel Streaming Response** node. Specify a custom string argument to the node to indicate to the player page what event has happened.  
+1.  In your Unreal Engine application, any time you want to emit an event to the player page, you need an actor with a **Pixel Streaming Input** component. This component has access to the **Send Pixel Streaming Response** node. Specify a custom string argument to the node to indicate to the player page what event has happened.
```
Comment (lukehb) [inline Frontend/Docs/Communicating from UE5 to the Player Page.md]: > to indicate to the player page what event has happened

to 

> to send a message to the web frontend.

**Review by lukehb — CHANGES_REQUESTED** [review]

File: Frontend/Docs/Communicating from UE5 to the Player Page.md (hunk 1 above)
Comment (StomyPX) [reply Frontend/Docs/Communicating from UE5 to the Player Page.md]: Done

File: Frontend/Docs/Customizing Player Input Options.md [hunk 2]
```diff
@@ -4,44 +4,42 @@ The frontend library exposes a number of configuration options through the [Conf
 
 The following options are available in the frontend library:
```
Comment (lukehb) [inline Frontend/Docs/Customizing Player Input Options.md]: > The following options are available in the frontend library:

to

> The following options are available in the frontend library to customize input:

File: Frontend/Docs/Customizing Player Input Options.md [hunk 3]
```diff
@@ -4,44 +4,42 @@ The frontend library exposes a number of configuration options through the [Conf
 
 The following options are available in the frontend library:
 
-**TO DO**: Update this information to match the new front end.
+|       Property       | Default | Description |
+|         ---          |   ---   |     ---     |
+|   HoveringMouseMode  |  false  | Determines whether or not the player widget captures and locks the mouse when the player interacts with the widget. When enabled, the mouse cursor hovers over the player widget without interacting with it. In order to send the mouse movements to the input controller of the Unreal Engine application, the user needs to click and hold the left button of the mouse. Otherwise, clicking on the player widget causes it to capture and lock the mouse cursor. Any further movements of the mouse are passed immediately to the input controller in the Unreal Engine application. This typically allows the user to move and rotate the camera by simply dragging the mouse. To release the cursor from the control of the player widget, the user can press the **Esc** key. |
```
Comment (lukehb) [inline Frontend/Docs/Customizing Player Input Options.md]: > player widget

to

> video element

File: Frontend/Docs/Customizing Player Input Options.md [hunk 4]
```diff
@@ -4,44 +4,42 @@ The frontend library exposes a number of configuration options through the [Conf
...
+|  SuppressBrowserKeys |   true  | When this setting is enabled, the player widget will intercept function keys (**F1** to **F12**) and the **Tab** key, and pass those keypress events through to the Unreal Engine application rather than allowing the browser to process them normally.| This means, for example, that while this setting is active, pressing **F5** will not refresh the player page in the browser. Instead, that event is passed through to the Unreal Engine application, and has its usual function of switching the view to visualize shader complexity.
+| FakeMouseWithTouches |  false  | When this option is enabled, and the user is viewing the stream on a device with a touch screen such as a smartphone or tablet, this setting causes single-finger touch events to be interpreted by the Unreal Engine application as mouse clicks and drag events. Enabling this setting can provide users on mobile devices with the ability to partially control your Unreal Engine application, even when the application's input controller does not specifically handle touch input events. |
```
Comment (lukehb) [inline Frontend/Docs/Customizing Player Input Options.md]: Remove comma from:

> When this option is enabled,

File: Frontend/Docs/Customizing Player Input Options.md [hunk 5]
```diff
@@ -4,44 +4,42 @@ The frontend library exposes a number of configuration options through the [Conf
...
-| fakeMouseWithTouches | false | When this option is enabled, and the user is viewing the stream on a device with a touch screen such as a smartphone or tablet, this setting causes single-finger touch events to be interpreted by the Unreal Engine application as mouse clicks and drag events. Enabling this setting can provide users on mobile devices with the ability to partially control your Unreal Engine application, even when the application's input controller does not specifically handle touch input events. |
+When creating a frontend implementation, these options are visible via the `Config` object required in order to create a `PixelStreaming` stream for your frontend application. Simply set the values you want before initializing the stream object.
```
Comment (lukehb) [inline Frontend/Docs/Customizing Player Input Options.md]: `Config` and `PixelStreaming` could link to the relevant source files in repo?

**Review by lukehb — CHANGES_REQUESTED** [review]
Left some feedback on the customizing input options page

File: Frontend/Docs/Communicating from the Player Page to UE5.md [hunk 6]
```diff
@@ -2,68 +2,87 @@
...
 
+[//]: # (This is still up to date as of the Typescript Frontend rewrite)
 <p align="center">
-    <img src="Resources\Images\pixelstreaming-uiinteractionrespond.JPG" alt="Bind Event to OnPixelStreamingInputEvent">
+    <img src="Resources/Images/pixelstreaming-ui-interaction-event.png" alt="Bind Event to On Input Event">
 </p>
 
-You need to bind this event once, typically at the start of your game. Each time any player HTML page connected to an instance of your Unreal Engine application calls the `emitUIInteraction`function, your custom event is automatically invoked, regardless of the input passed to `emitUIInteraction`.  
+You need to bind this event once, typically at the start of your game. Each time any player HTML page connected to an instance of your Unreal Engine application calls the `emitUIInteraction`function, your custom event is automatically invoked, regardless of the input passed to `emitUIInteraction`.
```
Comment (DenisTensorWorks) [inline Frontend/Docs/Communicating from the Player Page to UE5.md]: > application calls the `emitUIInteraction`function

to

> application calls the `emitUIInteraction` function

(note the space after Interaction' ; web version misinterprets the entire line as ' ')

File: Frontend/Docs/Customizing Player Input Options.md (hunk 2 above)
Comment (StomyPX) [reply Frontend/Docs/Customizing Player Input Options.md]: Done

File: Frontend/Docs/Customizing Player Input Options.md (hunk 3 above)
Comment (StomyPX) [reply Frontend/Docs/Customizing Player Input Options.md]: Done

File: Frontend/Docs/Customizing Player Input Options.md (hunk 4 above)
Comment (StomyPX) [reply Frontend/Docs/Customizing Player Input Options.md]: Done

File: Frontend/Docs/Customizing Player Input Options.md (hunk 5 above)
Comment (StomyPX) [reply Frontend/Docs/Customizing Player Input Options.md]: I've linked them for now to the absolute URLs on github. Is there a way to make them relative to the repo somehow?

File: Frontend/Docs/Communicating from the Player Page to UE5.md (hunk 6 above)
Comment (StomyPX) [reply Frontend/Docs/Communicating from the Player Page to UE5.md]: Good catch, done

File: Frontend/Docs/Customizing Player Input Options.md (hunk 5 above)
Comment (lukehb) [reply Frontend/Docs/Customizing Player Input Options.md]: Yeah, you should be able to.

For example see on the Docs page we do

`- [Customizing the Player Webpage](Docs/Customizing%20the%20Player%20Webpage.md)`

Comment (StomyPX) [reply Frontend/Docs/Customizing Player Input Options.md]: Okay, yeah I updated to these now.

File: Frontend/Docs/Customizing the Player Webpage.md [hunk 7]
```diff
@@ -1,21 +1,25 @@
...
-The git command mentioned above will pull the 5.1 branch of the infrastructure. If you need a different branch, please modify the git command accordingly.
+The git command mentioned above will pull the 5.2 branch of the infrastructure. If you need a different branch, please modify the git command accordingly.
+
+For more information about the Pixel Streaming front end and webserver changes, see [Pixel Streaming Infrastructure](https://docs.unrealengine.com/5.2/en-US/pixel-streaming-infrastructure/)
+
+We recommend starting with the [sample implementations](/Frontend/implementations/EpicGames/src) in order to judge how to put a new player page together and integrate it with your Unreal Engine application. Additionally, if you have cloned the Pixel Streaming Infrastructure repository and made upstream changes, you can fork the repo and make a pull request.
+
+
+## Customising the Player Webpage
+The Pixel Streaming Signaling and Web Server provides a sample player page that is already set up to stream in media from your Unreal Engine application and to send mouse, keyboard, and touch events back to the application. You can use this default player page as-is, if it meets your needs.
```
Comment (lukehb) [inline Frontend/Docs/Customizing the Player Webpage.md]: > Signaling

to

> Signalling

not sure why, but convention throughout source and docs is two l's

File: Frontend/Docs/HTML Page Requirements.md [hunk 8]
```diff
@@ -1,33 +1,39 @@
...
+*	You can create a new implementation page and place it in [`/Frontend/implementations/EpicGames/src/`](/Frontend/implementations/EpicGames/src) alongside the sample implementations. This must consist of both a base `.html` page and the `.ts` source for your application's entrypoint. This will then be accessible by appending the name of the `html` file to IP address or hostname of the computer running the Signalling Server.
+	For example, the sample `stresstest` page can be accessed on a locally-running infrastructure at `http:/127.0.0.1/stresstest.html`.
+*   You can customize the `HomepageFile` parameter for the Signaling and Web Server, and set the path to the filename of your custom HTML player page relative to the [Frontend implementations source folder](/Frontend/implementations/src). It will then be accessible when you access the IP address or hostname of the computer running the Signaling and Web Server.
```
Comment (lukehb) [inline Frontend/Docs/HTML Page Requirements.md:32]: > Signaling

to

> Signalling

**Review by lukehb — CHANGES_REQUESTED** [review]
Very good docs writing. Made some minor suggestions.

**Review by lukehb — APPROVED** [review]
Made the small changes myself. Approved.

**Pull request merged**